src/bindgen/builder.rs

```diff
@@ -394,6 +394,39 @@ impl Builder {
 
         result.source_files.extend_from_slice(self.srcs.as_slice());
 
+        // for (key, value) in &result.assoc_types{
+        //     println!("Key: {:#?}", key);
+        //     println!("Conc-type: {:#?}", value);
+        // }
+
+        // for func in result.functions.iter(){
+        //     println!("\nFunc: {:#?}\n", func);
+        // }
+
+        // Replace associated types in function signatures
+        for func in result.functions.iter_mut(){
+            // Check return type
+            if func.ret_assoc.is_some(){
+                let key = func.ret_assoc.as_ref().unwrap();
+                let concrete_type = result.assoc_types.get(key);
+                if concrete_type.is_some(){
+                    func.ret = concrete_type.unwrap().clone();
+                }
+            }
```

Comment:
```suggestion
            if let Some(key) = &func.ret_assoc {
                let concrete_type = result.assoc_types.get(key);
                if concrete_type.is_some(){
                    func.ret = concrete_type.unwrap().clone();
                }
            }
```

apply this elsewhere as well

Reply:
it's a bad design to have `Option::is_some` followed by `unwrap`. Having `unwrap` or `expect` is usually an indicator of bad code design

Reply:
you can either use `if let` or you can use `Option::map` if you prefer the functional style